src/java/org/apache/cassandra/transport/Dispatcher.java

```diff
@@ -191,7 +220,20 @@ else if (backpressure == Overload.BYTES_IN_FLIGHT)
 
         Message.logger.trace("Received: {}, v={}", request, connection.getVersion());
         connection.requests.inc();
-        Message.Response response = request.execute(qstate, startTimeNanos);
+        long cqlStartTimeNanos;
+        Config.CQLStartTime cqlStartTime = DatabaseDescriptor.getCQLStartTime();
+        switch (cqlStartTime)
+        {
+            case QUEUE:
+                cqlStartTimeNanos = request.createdAtNanos;
```

Comment:
While I'll support this change, it makes 
https://issues.apache.org/jira/browse/CASSANDRA-19330 very easy to hit.

Reply:
Similarly, it affects read repairs.